File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/observer/observer/ConcreteSubject.java
package com.bigboxcode.designpattern.observer.observer;

import java.util.ArrayList;
import java.util.List;

public class ConcreteSubject implements Subject{

    private int state;

    private List<Observer> observerList = new ArrayList<>();

    public int getState() {
        return state;
    }

    public void setState(int state) {
        this.state = state;

        notifyObservers();
    }

    public void attach(Observer observer) {
        observerList.add(observer);
    }

    public void notifyObservers() {
        for (Observer observer : observerList) {
            observer.sendUpdate();
        }
    }

}

File: design-patterns/php/observer/general/ConcreteSubject.php
<?php
// ConcreteSubject.php

namespace BigBoxCode\DesignPattern\Observer\General;

class ConcreteSubject implements Subject {
    private int $state = 0;
    private array $observerList = [];

    public function getState(): int {
        return $this->state;
    }

    public function setState(int $state): void {
        $this->state = $state;
        $this->notifyObservers();
    }

    public function attach(Observer $observer): void {
        array_push($this->observerList, $observer);
    }

    public function notifyObservers(): void {
        foreach ($this->observerList as $observer) {
            $observer->sendUpdate();
        }
    }
}